examples/acados_python/p_global_example/example_p_global.py

```diff
@@ -0,0 +1,299 @@
+
+    if lut:
+        # Coefficient of B-spline
+        C = MX.sym("C", data.shape[0], 1)
+        p_global += [C]
+        p_global_values = np.concatenate([p_global_values, data])
+    else:
+        C = None
+    p_global = ca.vcat(p_global)
+
+    return p_global, m, l, C, p_global_values
+
+
+def export_pendulum_ode_model(p_global, m, l, C, lut=True) -> AcadosModel:
+    model_name = 'pendulum'
+
+    # constants
+    m_cart = 1. # mass of the cart [kg]
+
+    # parameters
+    g = MX.sym("g")
+    p = g
+
+    # set up states & controls
+    x1      = MX.sym('x1')
+    theta   = MX.sym('theta')
+    v1      = MX.sym('v1')
+    dtheta  = MX.sym('dtheta')
+
+    x = vertcat(x1, theta, v1, dtheta)
+
+    F = MX.sym('F')
+    u = vertcat(F)
+
+    # xdot
+    nx = x.shape[0]
+    xdot = MX.sym('xdot', nx)
+
+    # dynamics
+    cos_theta = cos(theta)
+    sin_theta = sin(theta)
+    denominator = m_cart + m - m*cos_theta*cos_theta
+    f_expl = vertcat(v1,
+                     dtheta,
+                     (-m*l*sin_theta*dtheta*dtheta + m*g*cos_theta*sin_theta+F)/denominator,
+                     (-m*l*cos_theta*sin_theta*dtheta*dtheta + F*cos_theta+(m_cart+m)*g*sin_theta)/(l*denominator)
+                     )
+
+    if lut:
+        x_in = ca.vertcat(u/100+0.5,theta/np.pi+0.5)
+
+        # Disturb the dynamics by a sprinkle of bspline
+        f_expl[2:4] += 0.01*ca.bspline(x_in,C,knots,[3,2],2)
+
+    f_impl = xdot - f_expl
+
+    model = AcadosModel()
+
+    model.f_impl_expr = f_impl
+    model.f_expl_expr = f_expl
+    model.x = x
+    model.xdot = xdot
+    model.u = u
+    # model.z = z
+    model.p = p
+    model.p_global = p_global
+    model.name = model_name
+
+    return model
+
+
+def create_ocp_formulation_without_opts(p_global, m, l, C, lut=True, use_p_global=True) -> AcadosOcp:
+
+    # create ocp object to formulate the OCP
+    ocp = AcadosOcp()
+
+    # set model
+    model = export_pendulum_ode_model(p_global, m, l, C, lut=lut)
+    model.p_global = p_global
+    ocp.model = model
+
+    # dimensions
+    nx = model.x.rows()
+    nu = model.u.rows()
+    ny = nx + nu
+    ny_e = nx
+
+    # set cost
+    Q = 2*np.diag([1e3, 1e3, 1e-2, 1e-2])
+    R = 2*np.diag([1e-2])
+
+    ocp.cost.W_e = Q
+    ocp.cost.W = scipy.linalg.block_diag(Q, R)
+
+    ocp.cost.cost_type = 'LINEAR_LS'
+    ocp.cost.cost_type_e = 'LINEAR_LS'
+
+    ocp.cost.Vx = np.zeros((ny, nx))
+    ocp.cost.Vx[:nx,:nx] = np.eye(nx)
+
+    Vu = np.zeros((ny, nu))
+    Vu[4,0] = 1.0
+    ocp.cost.Vu = Vu
+
+    ocp.cost.Vx_e = np.eye(nx)
+
+    ocp.cost.yref  = np.zeros((ny, ))
+    ocp.cost.yref_e = np.zeros((ny_e, ))
+
+    # set constraints
+    Fmax = 80
+    ocp.constraints.lbu = np.array([-Fmax])
+    ocp.constraints.ubu = np.array([+Fmax])
+    ocp.constraints.idxbu = np.array([0])
+
+    ocp.constraints.x0 = np.array([0.0, np.pi, 0.0, 0.0])
+
+    ocp.parameter_values = np.array([9.81])
+
+    if not use_p_global:
```

Comment:
maybe move up to where `p_global` is set initially

Reply:
`p_global_values` are not available at this point. This split is done in the example to allow for testing an equivalent formulation without `p_global` and corresponding version with multi-phase.